### PR TITLE
Update to GHC 9.4 and `brick` 1.5

### PR DESCRIPTION
**Pull request by kostmo**
A couple upgrades:

- Upgrade to Brick 1.5, which has a new nicer dialog API, with names to identify buttons instead of positional indices: https://hackage.haskell.org/package/brick-1.5/changelog
- Update stack resolver to a nightly one, with GHC 9.4.  This fixes? mitigates? #1000, at least, no one has yet observed the issue under GHC 9.4 that we know of.

### CODE REVIEW

File: swarm.cabal [hunk 1]
```diff
@@ -147,7 +147,7 @@ library
     build-depends:    base                          >= 4.14 && < 4.18,
                       aeson                         >= 2 && < 2.2,
                       array                         >= 0.5.4 && < 0.6,
-                      brick                         >= 1.0 && < 1.5,
+                      brick                         >= 1.0 && < 1.6,
```
Comment (byorgey) [inline swarm.cabal:150]: Should this be `brick >= 1.5 && < 1.6`?  I assume it will not compile with earlier versions of `brick`, right?

**Comment by byorgey** [comment]
@kostmo do you have a sense of what would be required to finish this?  Can I help?  I'd love to get this merged since #1000 is very annoying.  It's easy to work around for more experienced developers but I'd really like to be able to continue telling people to `stack run` and have it just work.

File: swarm.cabal (hunk 1 above)
Comment (kostmo) [reply swarm.cabal:150]: correct

**Comment by kostmo** [comment]
> Can I help?

I'll happily hand this over to you if you want to finish it.  Mostly I think the work entails creating and assigning new members of the `Name` type to dialog buttons.

**Comment by byorgey** [comment]
OK, I'll take a look.

File: src/Swarm/TUI/Model/Name.hs [hunk 2]
```diff
@@ -16,8 +16,17 @@ data GoalWidget
   | GoalSummary
   deriving (Eq, Ord, Show, Read, Bounded, Enum)
 
+-- | Clickable buttons in modal dilaogs.
```
Comment (kostmo) [inline src/Swarm/TUI/Model/Name.hs:19]: "dialogs"

**Comment by xsebek** [comment]
@byorgey should we merge it before release?

**Review by xsebek — APPROVED** [review]
LGTM! 👍

**Pull request merged**